src/libs/antares/study/scenario-builder/rules.h

```diff
@@ -112,14 +111,30 @@ class Rules final : private Yuni::NonCopyable<Rules>
     hydroTSNumberData hydro;
     //! Wind
     windTSNumberData wind;
+
+    // gp : change these 2 arrays into std containers, like std::vector 
     //! Thermal (array [0..pAreaCount - 1])
     thermalTSNumberData* thermal;
     //! Renewable (array [0..pAreaCount - 1])
     renewableTSNumberData* renewable;
+
     //! hydro levels
     hydroLevelsData hydroLevels;
 
 private:
+    // Member methods
+    void readThermalCluster(const AreaName::Vector& instrs, String value, bool updaterMode);
+    void readRenewableCluster(const AreaName::Vector& instrs, String value, bool updaterMode);
+    void readLoad(const AreaName::Vector& instrs, String value, bool updaterMode);
+    void readWind(const AreaName::Vector& instrs, String value, bool updaterMode);
+    void readHydro(const AreaName::Vector& instrs, String value, bool updaterMode);
+    void readSolar(const AreaName::Vector& instrs, String value, bool updaterMode);
+    void readHydroLevels(const AreaName::Vector& instrs, String value, bool updaterMode);
+
+    Data::Area* getArea(const AreaName& areaname, bool updaterMode);
+
+    // Member data 
+    Study& study_;
```

Comment:
Set this member as protected, and re-use in classes `loadTSNumberData`, `hydroTSNumberData`, etc. `loadTSNumberData::apply`, `hydroTSNumberData::apply` will not need to be passed a reference to `study`.